NetKet/Machine/activations.hpp

```diff
@@ -0,0 +1,73 @@
+#include <Eigen/Dense>
+#include <complex>
+#include <iostream>
+#include <random>
+#include <vector>
+
+#ifndef NETKET_ACTIVATIONS_HPP
+#define NETKET_ACTIVATIONS_HPP
+
+namespace netket {
+
+class AbstractActivation {
+ public:
+  using VectorType = Eigen::Matrix<std::complex<double>, Eigen::Dynamic, 1>;
+
+  virtual inline void activate(const VectorType &Z, VectorType &A) = 0;
+  virtual inline void apply_jacobian(const VectorType &Z, const VectorType &A,
```

Comment:
Please include a virtual destructor to cure the memory leaks dedected in Travis

Reply:
added